services/queue/test/reruntask_test.js

```diff
@@ -35,32 +35,64 @@ helper.secrets.mockSuite(testing.suiteName(), ['aws'], function(mock, skipping)
       purpose: 'taskcluster-testing',
     },
   };
+  let taskId;
 
-  test('create, claim, complete and rerun (is idempotent)', async () => {
-    const taskId = slugid.v4();
+  setup(() => {
+    taskId = slugid.v4();
+  });
 
+  const createTask = async taskDef => {
     debug('### Creating task');
     await helper.queue.createTask(taskId, taskDef);
     helper.assertPulseMessage('task-defined');
-    helper.assertPulseMessage('task-pending', m => m.payload.runId === 0);
+    if (!taskDef.dependencies) {
+      helper.assertPulseMessage('task-pending', m => m.payload.runId === 0);
+    }
+  };
 
+  const claimTask = async () => {
     debug('### Claiming task');
     // First runId is always 0, so we should be able to claim it here
     await helper.queue.claimTask(taskId, 0, {
       workerGroup: 'my-worker-group-extended-extended',
       workerId: 'my-worker-extended-extended',
     });
     helper.assertPulseMessage('task-running');
+  };
+
+  test('rerun a scheduled task', async () => {
+    // create a self-dependent task, so it is in state 'scheduled'
+    await createTask({ ...taskDef, dependencies: [taskId] });
+
+    debug('### Requesting task rerun');
+    helper.scopes('queue:rerun-task:my-scheduler-extended-extended/dSlITZ4yQgmvxxAi4A8fHQ/*');
```

Comment:
How was this task originally created?
Do we need to account for it in any development docs?

Reply:
It's created on line 65 :)

Reply:
Ah, the `dslITZ..` slugId is the taskGroupId, which is hard-coded a bit further up.  None of this stuff runs against the production queue -- it's running against the queue under test.